### PR TITLE
*: make explain result more clearly.

### LINKED ISSUE #2026: Error when explain a query
Please answer these questions before submitting your issue. Thanks!

1. What version of Go are you using (`go version`)?
1.7.3

2. What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN="/usr/local/go/bin"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/gopkg"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build018490638=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"

3. What did you do?
If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
```
drop table if exists t2, t3;
create table t2(b int);
create table t3(a int);
explain select * from t3 where a >= all (select b from t2);
```
4. What did you expect to see?

5. What did you see instead?
`[error] json: error calling MarshalJSON for type *plan.Projection: json: error calling MarshalJSON for type *plan.Selection: json: error calling MarshalJSON for type *plan.PhysicalApply: invalid character '3' after array element`

### PR DESCRIPTION
**Pull request by hanfei1991**
Output the id in first column.
fix #2026 
@shenli @coocood @zimulala @tiancaiamao @XuHuaiYu PTAL

### CODE REVIEW

**Comment by shenli** [comment]
Should we update tidb-test?

**Comment by hanfei1991** [comment]
@shenli yes, we need

**Comment by coocood** [comment]
I think parent should appear before children.

**Comment by hanfei1991** [comment]
@coocood We have lost parents info after physical plan building. I will fix it in future.

File: executor/explain.go [hunk 1]
```diff
@@ -25,33 +25,56 @@ import (
...
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	e.cursor++
+	if e.cursor > len(e.rows) {
+		return nil, nil
 	}
-	return row, nil
+	return e.rows[e.cursor-1], nil
```
Comment (coocood) [inline executor/explain.go]:
```
row := e.rows[e.cursor]
cursor++ 
```

**Comment by coocood** [comment]
LGTM

File: plan/physical_plan_builder.go [hunk 2]
```diff
@@ -141,6 +141,9 @@ func (p *DataSource) convert2TableScan(prop *requiredProperty) (*physicalPlanInf
 		DBName:              p.DBName,
 		physicalTableSource: physicalTableSource{client: client},
 	}
+	ts.tp = "TableScan"
```
Comment (zimulala) [inline plan/physical_plan_builder.go]: Why not use the constants in plan/plan.go ? The same as other `tp`.

Comment (hanfei1991) [inline plan/physical_plan_builder.go]: I will remove the constants in future. The type constant won't be used frequently. It's better to use string literal.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**